test/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/ValidConnectionMethodBase64EncodedDetailsSpec.scala

```diff
@@ -24,28 +24,84 @@ import uk.gov.hmrc.hmrcemailrenderer.templates.tdq.params.{ HeadersValidation, P
 
 class ValidConnectionMethodBase64EncodedDetailsSpec extends WordSpec with Matchers {
 
-  "Extra details needed" should {
+  val requestCount = 100
+
+  "Details" should {
+
+    "have connection method in the same format as the specification" in {
+      val details = ValidConnectionMethodBase64EncodedDetails("WEB_APP_VIA_SERVER", requestCount, Set.empty)
+      details.prettyConnectionMethod shouldBe "Web application via server"
+    }
+  }
+
+  "Extra details" should {
+
+    "have a single header in the same format as the specification" in {
+      val validation = HeadersValidation("gov-client-public-ip", Set.empty, Set.empty)
+      validation.prettyHeaderOrHeaders shouldBe "Gov-Client-Public-IP"
+    }
+
+    "have multiple headers in the same format as the specification" in {
+      val validation =
+        HeadersValidation("gov-vendor-forwarded, gov-vendor-public-ip, gov-client-public-ip", Set.empty, Set.empty)
+      validation.prettyHeaderOrHeaders shouldBe "Gov-Vendor-Forwarded, Gov-Vendor-Public-IP, Gov-Client-Public-IP"
+    }
+
+    "display record count in human friendly format" in {
```

Comment:
There is a class called `ProblemSpec` where `percentageDescription` is unit tested. I expected to find new unit tests there as well for 1k stuff but it's not there. Since `percentDescription` and `prettyCount` are very similar conceptually maybe we could have tests for both of them in the same location be it `ProblemSpec` or `ValidConnectionMethodBase64EncodedDetailsSpec` - this I don't mind.

Reply:
Moved to `ProblemSpec` where it is probably better suited